Intersections_3/include/CGAL/Intersections_3/internal/Bbox_3_Triangle_3_do_intersect.h

```diff
@@ -179,50 +188,52 @@ namespace internal {
+                                    const std::array< std::array<FT, 3> , 3>& sides,
+                                    const Box3& bbox,
+                                    Fct do_axis_intersect_aux_impl)
   {
-    const typename K::Point_3* j = & triangle.vertex(SIDE);
-    const typename K::Point_3* k = & triangle.vertex((SIDE+2)%3);
+    // TODO: no longer a ptr, is it an issue for swap?
+    std::array<FT, 3> j = triangle[SIDE];
+    std::array<FT, 3> k = triangle[(SIDE+2)%3];
 
-    typename K::Point_3 p_min, p_max;
-    get_min_max<K,Box3, AXE>(AXE==0? 0: AXE==1? sides[SIDE].z(): -sides[SIDE].y(),
-                             AXE==0? -sides[SIDE].z(): AXE==1? 0: sides[SIDE].x(),
-                             AXE==0? sides[SIDE].y(): AXE==1? -sides[SIDE].x():
-                             typename K::FT(0), bbox, p_min, p_max);
+    std::array<FT, 3> p_min, p_max;
+    get_min_max<FT, Box3, AXE>(AXE==0? 0: AXE==1? sides[SIDE][2]: -sides[SIDE][1],
+                               AXE==0? -sides[SIDE][2]: AXE==1? 0: sides[SIDE][0],
+                               AXE==0? sides[SIDE][1]: AXE==1? -sides[SIDE][0]:
+                               FT(0), bbox, p_min, p_max);
 
     switch ( AXE )
     {
     case 0: {
       // t_max >= t_min
-      Uncertain<bool> b =  do_axis_intersect_aux<K,AXE,SIDE>(k->y()-j->y(), k->z()-j->z(), sides) >= 0;
+      Uncertain<bool> b =  do_axis_intersect_aux_A0<FT,SIDE>(k[1]-j[1], k[2]-j[2], sides, do_axis_intersect_aux_impl) != NEGATIVE;
       if (is_indeterminate(b))
         return b;
       if(b) std::swap(j,k);
-      return CGAL_AND((do_axis_intersect_aux<K,AXE,SIDE>(p_min.y()-j->y(), p_min.z()-j->z(), sides) <= 0),
-                      (do_axis_intersect_aux<K,AXE,SIDE>(p_max.y()-k->y(), p_max.z()-k->z(), sides) >= 0) );
+      return CGAL_AND((do_axis_intersect_aux_A0<FT,SIDE>(p_min[1]-j[1], p_min[2]-j[2], sides, do_axis_intersect_aux_impl) != POSITIVE),
+                      (do_axis_intersect_aux_A0<FT,SIDE>(p_max[1]-k[1], p_max[2]-k[2], sides, do_axis_intersect_aux_impl) != NEGATIVE) );
     }
     case 1: {
       // t_max >= t_min
-      Uncertain<bool> b =  do_axis_intersect_aux<K,AXE,SIDE>(k->x()-j->x(), k->z()-j->z(), sides) >= 0;
+      Uncertain<bool> b =  do_axis_intersect_aux_A1<FT,SIDE>(k[0]-j[0], k[2]-j[2], sides, do_axis_intersect_aux_impl) != NEGATIVE;
       if (is_indeterminate(b))
         return b;
       if(b) std::swap(j,k);
-      return  CGAL_AND((do_axis_intersect_aux<K,AXE,SIDE>(p_min.x()-j->x(), p_min.z()-j->z(), sides) <= 0),
-                       (do_axis_intersect_aux<K,AXE,SIDE>(p_max.x()-k->x(), p_max.z()-k->z(), sides) >= 0) );
+      return  CGAL_AND((do_axis_intersect_aux_A1<FT,SIDE>(p_min[0]-j[0], p_min[2]-j[2], sides, do_axis_intersect_aux_impl) != POSITIVE),
+                       (do_axis_intersect_aux_A1<FT,SIDE>(p_max[0]-k[0], p_max[2]-k[2], sides, do_axis_intersect_aux_impl) != NEGATIVE) );
 
```

Comment:
Don't we have to write
````
b = Uncertain<bool>::indeterminate();
      try {
        b = CGAL_AND((do_axis_intersect_aux_A1<FT,SIDE>(p_min[0]-j[0], p_min[2]-j[2], sides, do_axis_intersect_aux_impl) != POSITIVE),
                       (do_axis_intersect_aux_A1<FT,SIDE>(p_max[0]-k[0], p_max[2]-k[2], sides, do_axis_intersect_aux_impl) != NEGATIVE) );
      }catch(const Uncertain_conversion_exception& ){
      }
      return b;
```

Reply:
Why?
(by the way, now that we require C++14, I think it would be a good time to rewrite CGAL_AND in Uncertain.h without statement expressions so visual studio can benefit as well)

Reply:
@mglisse started in #5904

Reply:
For the record, Andreas thought that a try/catch was missing because some parts of the predicate unprotect in the static version of the predicate and that was an issue on windows because `CGAL_AND` was converting the `Uncertain<bool>` to `bool`.